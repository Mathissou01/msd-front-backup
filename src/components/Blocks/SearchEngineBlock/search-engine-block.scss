@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/02-tools/tools.inputs" as input;
@use "src/styles/02-tools/tools.include-media" as im;

.c-SearchEngineBlock {
  display: flex;
  flex-direction: column;
  gap: 32px;
  align-items: center;
  padding: 40px 16px;
  background-color: colors.$white;

  @include im.media(">=tablet") {
    padding: 60px 160px 40px;
  }

  @include im.media(">=laptop") {
    padding: 80px 160px 40px;
  }

  @include im.media(">=desktop") {
    padding: 80px 124px 40px;
  }

  @include im.media(">=large-desktop") {
    padding: 80px 224px 40px;
  }

  &__Heading,
  &__SearchContainer {
    display: flex;
    flex-direction: column;
    width: 100%;
    max-width: 648px;

    @include im.media(">=tablet") {
      max-width: 703px;
    }

    @include im.media(">=desktop") {
      max-width: 785px;
    }

    @include im.media(">=large-desktop") {
      max-width: 945px;
    }
  }

  &__Title {
    @include typo.text;

    color: colors.$expert-blue;
    font-size: 18px;
    text-align: center;
  }

  &__Subtitle {
    @include typo.heading(2);

    color: colors.$expert-blue;
    text-align: center;
  }

  &__SearchContainer {
    gap: 12px;
  }

  &__SearchBar {
    display: flex;
    flex-direction: column;
    gap: 8px;

    @include im.media(">=mobile") {
      flex-direction: row;

      button {
        width: unset;
      }
    }
  }

  &__Input {
    @include input.input;

    height: 56px;
  }
}
