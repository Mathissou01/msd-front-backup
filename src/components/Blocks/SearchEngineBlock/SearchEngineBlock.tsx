import React, { useEffect, useState } from "react";
import CommonButton from "../../Common/CommonButton/CommonButton";
import "./search-engine-block.scss";

interface ISearchEngineBlock {
  searchTerm: string;
  title?: string;
  subtitle?: string;
  placeholder?: string;
  minLengthSearch?: number;
  onSearchTermChange: (searchTerm: string) => void;
  onSearchSubmit: () => void;
}

export default function SearchEngineBlock({
  searchTerm,
  title,
  subtitle,
  placeholder,
  minLengthSearch = 3,
  onSearchTermChange,
  onSearchSubmit,
}: ISearchEngineBlock) {
  const [isDisabled, setIsDisabled] = useState(true);

  const labels = {
    buttonLabel: "Rechercher",
  };

  useEffect(() => {
    setIsDisabled(searchTerm.length < minLengthSearch);
  }, [searchTerm, minLengthSearch]);

  return (
    <section className="c-SearchEngineBlock">
      {title !== "" || subtitle !== "" ? (
        <div className="c-SearchEngineBlock__Heading">
          {title !== "" ? (
            <span className="c-SearchEngineBlock__Title">{title}</span>
          ) : null}
          {subtitle !== "" ? (
            <h2 className="c-SearchEngineBlock__Subtitle">{subtitle}</h2>
          ) : null}
        </div>
      ) : null}

      <div className="c-SearchEngineBlock__SearchContainer">
        <div className="c-SearchEngineBlock__SearchBar">
          <input
            className="c-SearchEngineBlock__Input"
            placeholder={placeholder}
            type="text"
            onChange={(e) => onSearchTermChange(e.target.value)}
            aria-label="Rechercher un dÃ©chet"
          />
          <CommonButton
            label={labels.buttonLabel}
            picto="search"
            type="button"
            style="primary"
            fontStyle="fontLarge"
            paddingStyle="paddingLarge"
            isFullWidth={true}
            isDisabled={isDisabled}
            onClick={onSearchSubmit}
          />
        </div>
      </div>
    </section>
  );
}
