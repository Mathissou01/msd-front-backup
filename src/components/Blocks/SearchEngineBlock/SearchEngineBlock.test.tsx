import { fireEvent, render, screen } from "@testing-library/react";
import SearchEngineBlock from "./SearchEngineBlock";
import React from "react";

describe("SearchEngineBlock", () => {
  const data = {
    title: "Titre",
    subtitle: "Sous-titre",
    placeholder: "Lampes",
  };

  it("renders the block properly with all props", () => {
    const onSearchSubmit = jest.fn();
    const handleSearchTermChange = jest.fn();

    const { container } = render(
      <SearchEngineBlock
        title={data.title}
        subtitle={data.subtitle}
        placeholder={data.placeholder}
        onSearchSubmit={onSearchSubmit}
        onSearchTermChange={handleSearchTermChange}
        minLengthSearch={3}
        searchTerm=""
      />,
    );

    /* Components (input & button) */
    const input = screen.getByPlaceholderText("Lampes");
    const searchButton = screen.getByRole("button");

    /* Testing cases */
    expect(searchButton).toBeDisabled();
    expect(input).toHaveValue("");
    expect(container).toMatchSnapshot();
  });

  it("should modify input value by firing onChangeEvent", () => {
    const handleSearchTermChange = jest.fn();

    const { container } = render(
      <SearchEngineBlock
        title={data.title}
        subtitle={data.subtitle}
        placeholder={data.placeholder}
        onSearchSubmit={() => null}
        onSearchTermChange={handleSearchTermChange}
        minLengthSearch={3}
        searchTerm=""
      />,
    );

    /* Components (input & button) */
    const input = screen.getByPlaceholderText("Lampes");
    const searchButton = screen.getByRole("button");

    /* Testing cases */
    expect(searchButton).toBeDisabled();
    fireEvent.change(input, { target: { value: "Test Input" } });
    expect(handleSearchTermChange).toBeCalledWith("Test Input");

    expect(container).toMatchSnapshot();
  });

  it("should firing click button 1 times when input value > 3", () => {
    const onSearchSubmit = jest.fn();
    const handleSearchTermChange = jest.fn();

    render(
      <SearchEngineBlock
        title={data.title}
        subtitle={data.subtitle}
        placeholder={data.placeholder}
        onSearchSubmit={onSearchSubmit}
        onSearchTermChange={handleSearchTermChange}
        minLengthSearch={3}
        searchTerm="Test Input"
      />,
    );

    /* Components (input & button) */
    const searchButton = screen.getByRole("button");

    /* Testing cases */
    // Button is enabled because it have value
    expect(searchButton).toBeEnabled();

    // We test click button
    fireEvent.click(searchButton);
    expect(onSearchSubmit).toBeCalledTimes(1);
  });
});
