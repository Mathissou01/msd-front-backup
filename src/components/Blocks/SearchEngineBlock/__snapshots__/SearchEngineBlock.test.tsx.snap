// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchEngineBlock renders the block properly with all props 1`] = `
<div>
  <section
    class="c-SearchEngineBlock"
  >
    <div
      class="c-SearchEngineBlock__Heading"
    >
      <span
        class="c-SearchEngineBlock__Title"
      >
        Titre
      </span>
      <h2
        class="c-SearchEngineBlock__Subtitle"
      >
        Sous-titre
      </h2>
    </div>
    <div
      class="c-SearchEngineBlock__SearchContainer"
    >
      <div
        class="c-SearchEngineBlock__SearchBar"
      >
        <input
          aria-label="Rechercher un déchet"
          class="c-SearchEngineBlock__Input"
          placeholder="Lampes"
          type="text"
        />
        <button
          class="c-CommonButton c-CommonButton_primary c-CommonButton_fontLarge c-CommonButton_paddingLarge c-CommonButton_fullWidth c-CommonButton_disabled"
          disabled=""
          type="button"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_search"
          />
          Rechercher
        </button>
      </div>
    </div>
  </section>
</div>
`;

exports[`SearchEngineBlock should modify input value by firing onChangeEvent 1`] = `
<div>
  <section
    class="c-SearchEngineBlock"
  >
    <div
      class="c-SearchEngineBlock__Heading"
    >
      <span
        class="c-SearchEngineBlock__Title"
      >
        Titre
      </span>
      <h2
        class="c-SearchEngineBlock__Subtitle"
      >
        Sous-titre
      </h2>
    </div>
    <div
      class="c-SearchEngineBlock__SearchContainer"
    >
      <div
        class="c-SearchEngineBlock__SearchBar"
      >
        <input
          aria-label="Rechercher un déchet"
          class="c-SearchEngineBlock__Input"
          placeholder="Lampes"
          type="text"
        />
        <button
          class="c-CommonButton c-CommonButton_primary c-CommonButton_fontLarge c-CommonButton_paddingLarge c-CommonButton_fullWidth c-CommonButton_disabled"
          disabled=""
          type="button"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_search"
          />
          Rechercher
        </button>
      </div>
    </div>
  </section>
</div>
`;
